src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -91,6 +91,12 @@ private static function compilePattern(Route $route, $pattern, $isHost)
         $matches = array();
         $pos = 0;
         $defaultSeparator = $isHost ? '.' : '/';
+        $useUnicode = preg_match('//u', $pattern);
+        $needsUnicode = $route->getUnicode() || ($useUnicode && preg_match('/[\x80-\xFF]/', $pattern));
```

Comment:
That's why I added "seamless" to the description. From the doc PR:

    Since Symfony 3.2, requirements that use any non-ASCII UTF-8 characters or a
    a `PCRE Unicode property`_ (``\p{xx}``, ``\P{xx}`` or ``\X``) turn on the
    `unicode` parameter on its corresponding route. You can also set this
    `unicode` parameter to true explicitly when declaring or importing routes so
    that e.g. a `.` matches any UTF-8 characters instead of just a single byte. 

This required to reduce the boilerplate. People want to write `[éè]` and have it work out of the box.